lms/djangoapps/program_enrollments/models.py

```diff
@@ -61,3 +63,29 @@ def retire_user(cls, user_id):
 
         enrollments.update(external_user_key=None)
         return True
+
+
+class ProgramCourseEnrollment(TimeStampedModel):  # pylint: disable=model-missing-unicode
+    """
+    This is a model to represent a learner's enrollment in a course
+    in the context of a program from the registrar service
+
+    .. no_pii:
+    """
+    STATUSES = (
+        ('active', 'active'),
+        ('inactive', 'inactive'),
+    )
+
+    class Meta(object):
+        app_label = "program_enrollments"
+
+    program_enrollment = models.ForeignKey(ProgramEnrollment, on_delete=models.CASCADE)
+    course_enrollment = models.OneToOneField(
+        StudentCourseEnrollment,
+        null=True,
+        blank=True,
+    )
+    course = models.ForeignKey(CourseOverview)
```

Comment:
I have concerns referencing CourseOverview like this. Did you get this way of storing course reference from the `CourseEnrollment` model? That might not be the right example to follow.
Everywhere within the `lms` namespace like `grades` and `course_goals` (https://github.com/edx/edx-platform/blob/master/lms/djangoapps/course_goals/models.py#L32), we use `CourseKeyField`. I think we need to do the same here

Reply:
Yes, I did get this from CourseEnrollment. Thanks for the heads up.